src/main/frontend/pipeline-console-view/pipeline-console/main/ConsoleLogCard.tsx

```diff
@@ -0,0 +1,195 @@
+import * as React from "react";
+import { styled } from "@mui/material/styles";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import CardActionArea from "@mui/material/CardActions";
+
+import Collapse from "@mui/material/Collapse";
+import IconButton, { IconButtonProps } from "@mui/material/IconButton";
+import Typography from "@mui/material/Typography";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import { StepInfo } from "./PipelineConsoleModel";
+import { ConsoleLine } from "./ConsoleLine";
+import Grid from "@mui/material/Grid";
+import Button from "@mui/material/Button";
+
+import { LOG_FETCH_SIZE } from "./PipelineConsoleModel";
+
+export interface StepSummaryProps {
+  step: StepInfo;
+}
+
+interface ConsoleLogCardProps {
+  step: StepInfo;
+  isExpanded: boolean;
+  handleStepToggle: (event: React.SyntheticEvent<{}>, nodeId: string) => void;
+  handleMoreConsoleClick: (nodeId: string, startByte: number) => void;
+}
+
+interface ExpandMoreProps extends IconButtonProps {
+  expand: boolean;
+}
+
+const ExpandMore = styled((props: ExpandMoreProps) => {
+  const { expand, ...other } = props;
+  return <IconButton {...other} />;
+})(({ theme, expand }) => ({
+  transform: !expand ? "rotate(0deg)" : "rotate(180deg)",
+  marginLeft: "auto",
+  transition: theme.transitions.create("transform", {
+    duration: theme.transitions.duration.shortest,
+  }),
+}));
+
+export class ConsoleLogCard extends React.Component<ConsoleLogCardProps> {
+  constructor(props: ConsoleLogCardProps) {
+    super(props);
+    this.handleStepToggle = this.handleStepToggle.bind(this);
+  }
+
+  handleStepToggle(event: React.MouseEvent<HTMLElement>) {
+    this.props.handleStepToggle(event, String(this.props.step.id));
+  }
+
+  getTrucatedLogWarning() {
+    if (this.props.step.consoleStartByte != 0) {
+      return (
+        <Grid container xs={12}>
+          <Grid item xs={6} sm>
+            <Typography align="right" className="step-header">
+              {`Missing ${this.prettySizeString(
+                this.props.step.consoleStartByte
+              )} of logs.`}
+            </Typography>
```

Comment:
I'm unsure if we need this text here, it would be good to be able to few the full steps log in an external page though.

Reply:
That's something I hope can be added by adding a hamburger menu to each step (with actions that can be taken on the step, like opening the full log in a new window - the API now supports this).
I added the text to make it clear how much text was missing. I would have liked lines, but that would require parsing all the lines upfront to get that information. At that point, you might as well return them and let the front end paginate them.

Reply:
I can make the text show a percentage of that's more meaningful?
We could also consider passing  the whole log back, as it only takes 2 seconds to parse 24MB of log. I would want that to be a separate change (with maybe an option to toggle it - to select a threshold from a list so people on slower networks can select to get less data).

Reply:
percentage would probably be better, but i'm not really sure if it's needed at all, could maybe be a tooltip over the Show more logs

Reply:
I can remove it if it's a problem. Just thought that if I had missing logs I'd want to know how much is missing.